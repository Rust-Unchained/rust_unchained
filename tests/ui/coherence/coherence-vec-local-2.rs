//@ aux-build:coherence_concrete_lib.rs

extern crate coherence_concrete_lib as lib;
use lib::Remote;

struct Local<T>(T);

// lib implements Remote for Vec<T: Clone>
impl<T> Remote for Vec<Local<T>> { }
//~^ ERROR E0119

// Additional requirement for failure in Unchained Rust,
// this creates a conflicting impl since lib implements Remote for Vec<T: Clone>
impl<T> Clone for Local<T> {
    fn clone(&self) -> Self {
        todo!()
    }
}

fn main() { }