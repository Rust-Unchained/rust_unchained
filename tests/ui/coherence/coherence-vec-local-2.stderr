error[E0119]: conflicting implementations of trait `Remote` for type `Vec<Local<_>>`
  --> $DIR/coherence-vec-local-2.rs:9:1
   |
LL | impl<T> Remote for Vec<Local<T>> { }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: conflicting implementation in crate `coherence_concrete_lib`:
           - impl<T> Remote for Vec<T>
             where T: Clone;

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0119`.
